src/components/Form/ExitFormDialogProvider.tsx

```diff
@@ -35,7 +37,9 @@ export const ExitFormDialogContext = React.createContext<ExitFormDialogData>({
   setEnableExitDialog: () => undefined,
   setExitDialogSubmitRef: () => undefined,
   shouldBlockNavigation: () => false,
-  setIsSubmitting: () => undefined
+  setIsSubmitting: () => undefined,
+  submit: () => Promise.resolve([]),
```

Comment:
Can we just skip the entire thing and use a check in useExitFormDialogProvider to make it strictly typed?